packages/react/src/components/Authenticator/hooks/useAuthenticator/index.tsx

```diff
@@ -125,9 +137,17 @@ export const useAuthenticator = (selector?: Selector) => {
      * We only trigger re-render if any of values in specified selected
      * values change. First compute the facade for prev and next state.
      */
-    const prevFacade = getFacade(prevState);
+    const prevFacade = prevFacadeRef.current;
```

Comment:
So the previous state is just a ref to the current state? Since the next state keeps changing?

Reply:
On each state transition, this comparator function gets called with `prevState` and `nextState`. `prevState` is the previous authenticator machine xstate state values. It'll look something like this:

```tsx
{
  value: 'signIn.runActor',
  context: {
    user: {...},
    actorRef: {...},
  },
   ...
}
```

While static / non-ref context values like `prevState.value` will be compared fine, static references like `actorRef` would always be considered equal even if its value in the ref has changed. That led to this code change.